controlplane/src/core/repositories/OrganizationRepository.ts

```diff
@@ -597,4 +605,246 @@ export class OrganizationRepository {
       await db.delete(targets).where(eq(targets.organizationId, organizationID)).execute();
     });
   }
+
+  public async createIntegration(input: {
+    organizationId: string;
+    endpoint: string;
+    name: string;
+    type: string;
+    events: string[];
+    eventsMeta: EventMeta[];
+  }) {
+    await this.db.transaction(async (db) => {
+      switch (input.type) {
+        case 'slack': {
+          const createSlackIntegrationResult = await db
+            .insert(organizationIntegrations)
+            .values({
+              organizationId: input.organizationId,
+              name: input.name,
+              type: 'slack',
+              events: input.events,
+            })
+            .returning()
+            .execute();
+
+          const slackIntegrationConfig = await db
+            .insert(slackIntegrationConfigs)
+            .values({
+              integrationId: createSlackIntegrationResult[0].id,
+              endpoint: input.endpoint,
+            })
+            .returning()
+            .execute();
+
+          for (const eventMeta of input.eventsMeta) {
+            switch (eventMeta.meta.case) {
+              case 'federatedGraphSchemaUpdated': {
+                const ids = eventMeta.meta.value.graphIds;
+                await db.insert(slackIntegrationEventConfigs).values({
+                  event: eventMeta.meta.case,
+                  slackIntegrationConfigId: slackIntegrationConfig[0].id,
+                  federatedGraphIds: ids || [],
+                });
+                break;
+              }
+            }
+          }
+        }
+      }
+    });
+  }
+
+  public async getIntegrationByName(organizationId: string, integrationName: string): Promise<Integration | undefined> {
```

Comment:
Should we maintain a postgres enum or at least typescript enum for the integration types?

Reply:
we do maintain a postgres enum for integration types

Reply:
Can we leverage it here?

Reply:
Done